app/controllers/users_controller.rb

```diff
@@ -1,3 +1,28 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find_by id: params[:id]
+    return if @user
+    flash[:danger] = t "users.user_not_found"
+    redirect_to signup_path
```

Comment:
You can redirect to root_path, because the user who is looking for other user may be signed in/signed up. :)